b2sdk/_internal/scan/folder.py

```diff
@@ -252,6 +252,20 @@ def _walk_relative_paths(
             local_path = local_dir / name
             relative_file_path = join_b2_path(relative_dir_path, name)
 
+            try:
+                name.encode('utf-8')
+            except UnicodeEncodeError:
+                if reporter is not None:
+                    reporter.invalid_unicode_filename(str(local_path))
+                continue
+
+            try:
+                validate_b2_file_name(name)
+            except ValueError as e:
+                if reporter is not None:
+                    reporter.invalid_filename(str(local_path), str(e))
+                continue
```

Comment:
```suggestion
            try:
                validate_b2_file_name(name)
            except ValueError as e:
                if reporter is not None:
                    reporter.invalid_filename(str(local_path), str(e))
                continue
```

Reply:
The only additional case I see that should have been accounted for, but wasn't is skipping directories with non-UTF8 encodable names